src/Swift.Bindings/src/TypeDatabase/TypeDatabase.cs

```diff
@@ -1,39 +1,73 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using System.Collections.Concurrent;
+using System.Diagnostics.CodeAnalysis;
 using System.Xml;
 
 namespace BindingsGeneration
 {
+
     /// <summary>
     /// Manages a mapping database between Swift types and C# types.
     /// </summary>
-    public unsafe class TypeDatabase
+    public class TypeDatabase : ITypeDatabase
```

Comment:
Does TypeDatabase always contain all modules? If yes, I would suggest:

```suggestion
    public static class TypeDatabase : ITypeDatabase
```

Reply:
I would say it depends on how we implement module by module processing algorithm, I would defer this until then